### ISSUE TITLE
BPM/tempo lock via MIDI?

### ISSUE THREAD

**Issue by mxmilkiib** [mixxxdj/mixxx#13038]
### Feature Description

See also #9978 #13031

**Comment by FrankwaP** [comment]
Hey :-)  I closed [my recent PR](https://github.com/mixxxdj/mixxx/pull/13025) as it was a duplicate of yours… but as I said creating controls for locking/unlocking the tempo was my second objective, and here there are: https://github.com/FrankwaP/mixxx/tree/tempo_locking_controls

I've created shortcuts and they work. I've also checked that they were displayed in the MIDI wizard and they are.

One annoying thing I've noticed however: when calling these when the channel is empty a segmentation fault! And this happens quite fast when one thinks these controls apply to the selected track in the library and forget to load it.  
I don't know if it's possible to do something smooth like the actual behavior of the right-click menu, which works on both the library and the channels…

**Comment by FrankwaP** [comment]
I've modified my code so it's closer to yours, but my controls still crash Mixxx when the Channel is empty.

Isn't it why we have these lines in the slot functions ?
```cpp
    if (v <= 0) {
        return;
    }
```

**Comment by ronso0** [comment]
> crash Mixxx when the Channel is empty

Sounds like you want to set beats on a track while the classes TrackPointer is null.
Just add null check like everywhere else in functions that operate on track objects.

**Comment by FrankwaP** [comment]
[Aaaaaaand done!](https://github.com/mixxxdj/mixxx/commit/5c37d2d2ad95bfea1cb4a209541d2249f07cd578) Than you for the pointer, @ronso0 :-)

Do I make a PR ?


BTW I have fetched/merged @mxmilkiib 's work locally and made a specific/minimalist keyboard mapping for tracks preparation:
```cfg
LoadSelectedTrack Shift+Left

cue_default f
play d

beats_set_halve F1
beats_set_twothirds F2
beats_set_threehalves F3
beats_set_double F4

beats_translate_earlier a
beats_translate_later z
unlock_tempo e
lock_tempo r

beatjump_backward q
beatjump_1_backward Shift+q
beatjump_forward s
beatjump_1_forward Shift+s

hotcue_1_activate w
hotcue_2_activate x
hotcue_3_activate c
hotcue_4_activate v
hotcue_5_activate Ctrl+w
hotcue_6_activate Ctrl+x
hotcue_7_activate Ctrl+c
hotcue_8_activate Ctrl+v
hotcue_1_clear Shift+w
hotcue_2_clear Shift+x
hotcue_3_clear Shift+c
hotcue_4_clear Shift+v
hotcue_5_clear Ctrl+Shift+w
hotcue_6_clear Ctrl+Shift+x
hotcue_7_clear Ctrl+Shift+c
hotcue_8_clear Ctrl+Shift+v
```

I just had a preparation session, and it goes faaaast <3

**Comment by ronso0** [comment]
> Do I make a PR ?

Yes please!

**Comment by ronso0** [comment]
Implemented by #13041
